### PR TITLE
Adding docs for KillSwitch #20265

### PR DESCRIPTION
**Pull request by svezfaz**
Refs #20265 
(also relates to #20441)

### CODE REVIEW

**Comment by ktoso** [comment]
OK TO TEST

**Comment by johanandren** [comment]
LGTM

File: akka-docs/rst/scala/code/docs/stream/KillSwitchDocSpec.scala [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      //#unique-shutdown
+      val countingSrc = Source(Stream.from(1)).delay(1.second, DelayOverflowStrategy.backpressure)
+      val lastSnk = Sink.last[Int]
+
+      val (killSwitch, last) = countingSrc
+        .viaMat(KillSwitches.single)(Keep.right)
+        .toMat(lastSnk)(Keep.both)
+        .run()
+
+      Thread.sleep(2500)
```
Comment (patriknw) [inline akka-docs/rst/scala/code/docs/stream/KillSwitchDocSpec.scala]: That is not so nice, and is also adding to the overall test execution time.
Can we just have `doSomethingElse()` and not verify the result in this doc test?

File: akka-docs/rst/scala/code/docs/stream/KillSwitchDocSpec.scala [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      val last = countingSrc
+        .via(sharedKillSwitch.flow)
+        .runWith(lastSnk)
+
+      val delayedLast = countingSrc
+        .delay(1.second, DelayOverflowStrategy.backpressure)
+        .via(sharedKillSwitch.flow)
+        .runWith(lastSnk)
+
+      Thread.sleep(2500)
```
Comment (patriknw) [inline akka-docs/rst/scala/code/docs/stream/KillSwitchDocSpec.scala]: same here

**Comment by patriknw** [comment]
Thanks for contributing @svezfaz 
Can you also add corresponding Java example and documentation?

File: akka-docs/rst/scala/code/docs/stream/KillSwitchDocSpec.scala (hunk 1 above)
Comment (johanandren) [inline akka-docs/rst/scala/code/docs/stream/KillSwitchDocSpec.scala]: For non-verifying test cases, check out `CompileOnlySpec`

**Comment by drewhk** [comment]
LGTM, but needs Java side docs and samples, too.

**Comment by svezfaz** [comment]
Added Java docs and samples + marked samples as compileOnly

File: akka-docs/rst/scala/code/docs/stream/KillSwitchDocSpec.scala (hunk 1 above)
Comment (patriknw) [inline akka-docs/rst/scala/code/docs/stream/KillSwitchDocSpec.scala]: @svezfaz can you fix this, thanks?

**Comment by drewhk** [comment]
Great, only the `sleep`s need to be removed now and it can go in.

**Comment by svezfaz** [comment]
Not sure if there is any agreed standard here, I put some dummy `doSomethingElse()` calls as suggested.

File: akka-stream/src/main/scala/akka/stream/KillSwitch.scala [hunk 3]
```diff
@@ -142,6 +143,7 @@ trait KillSwitch {
    */
   def abort(ex: Throwable): Unit
 }
+//#kill-switch
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/KillSwitch.scala]: a bit iffy on showing scala snippet in java documentation hm

File: akka-docs/rst/java/stream/stream-dynamic.rst [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+.. _kill-switch-scala:
+
+Controlling graph completion with KillSwitch
+--------------------------------------------
+
+A ``KillSwitch`` allows the completion of graphs of ``FlowShape`` from the outside. It consists of a flow element that
+can be linked to a graph of ``FlowShape`` needing completion control.
+The ``KillSwitch`` trait allows to complete or fail the graph(s).
+
+.. includecode:: ../../../../akka-stream/src/main/scala/akka/stream/KillSwitch.scala
```
Comment (ktoso) [inline akka-docs/rst/java/stream/stream-dynamic.rst]: Maybe don't include snippet since it's scala?

**Comment by ktoso** [comment]
Looks like this can go in today, LGTM - we can fix the scala code being shown in a java doc later

**Comment by svezfaz** [comment]
Thanks for the feedback. I'll  be able to fix this later today, if it can wait a few hours. Otherwise I can put in a follow-up change.

**Comment by johanandren** [comment]
I'll merge this and we can improve in a new PR for next version.

**Pull request merged**